strawberry/permission.py

```diff
@@ -151,9 +238,14 @@ def resolve(
         Checks if the permission should be accepted and
         raises an exception if not
         """
-        for permission in self.permissions:
-            if not permission.has_permission(source, info, **kwargs):
-                return self._on_unauthorized(permission)
+        try:
+            for permission in self.permissions:
+                permission.resolve_permission_sync(source, info, **kwargs)
+        except BaseException as e:
+            if self.fail_silently:
+                return [] if self.return_empty_list else None
+            else:
+                raise e
```

Comment:
Why did you remove the on_onauthorized call here? We also want to avoid catching all errors here. There may be some unexpected exceptions unrelated to permissions resulting from the has_permission call.

Reply:
I probably should have put a comment in the code for this:
BaseException was a placeholder, wasn't sure what exception type to catch, and the try catch is a bit of a performance hit anyway. If we can find a way to implement the composites while keeping the on_unauthorized call in the extension, this would naturally fall out nicely!